src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -565,9 +560,7 @@ def build(
             api_calls=select_child.api_calls,
             df_aliased_col_name_to_real_col_name=child.df_aliased_col_name_to_real_col_name,
             session=self.session,
-            referenced_ctes=child.referenced_ctes
```

Comment:
we probably also need to propagate the cte for the create_and_insert_plan https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py#L972 now need to have the cte propogated. 

For the subquery, we also need to set set the original cte in the newly resolved snowflake plan https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py#L326 which seems missing

Reply:
good catch.